mempool/v0/clist_mempool.go

```diff
@@ -183,6 +183,16 @@ func (mem *CListMempool) TxsFront() *clist.CElement {
 	return mem.txs.Front()
 }
 
+// GetTxByKey is not supported by the v0 mempool but it is required to satisfy the mempool interface.
```

Comment:
[question] why is it difficult to add support for GetTxByKey to the v0 mempool? Isn't this just a wrapper around the already existing `GetTx`?

Reply:
[blocking] I don't think we have rough consensus on deprecating the v0 mempool. See https://github.com/celestiaorg/celestia-core/issues/1382.

IMO if it's easy to add, then we should add support for `GetTxByKey` until we make a decision on deprecating the v0 mempool. If it's difficult to add support for `GetTxByKey` to the v0 mempool, that should be information that we use when making the decision to deprecate or not. If that's the case, making the decision should block merging this PR.